pkg/operator/resource/resourceapply/event_helpers.go

```diff
@@ -4,15 +4,33 @@ import (
 	"fmt"
 
 	"github.com/golang/glog"
-
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
+// getObjectKind returns a human name for the passed runtime object.
+// TODO: This won't handle openshift objects. For those objects we will have to pass the openshift/client-go scheme.
+func getObjectKind(object runtime.Object) (string, error) {
+	objectKind := "Object"
```

Comment:
try in this order:

1. GetObjectKind
2. if unstructured, search for kind and group in fields
3. then try kube scheme
4. then try openshift/api scheme
6. prepare for sadness when it doesn't work well